[BUG] arm  go test fail

### ✅ Verification Checklist

- [x] 🔍 I have searched the [existing issues](https://github.com/apache/dubbo-go-pixiu/issues) and confirmed this is not a duplicate

### 🚀 Go Version

go version go1.25.2 darwin/arm64

### 📦 Dubbo-go-pixiu Version

1.1.0

### 🖥️ Server Configuration

_No response_

### 💻 Client Configuration

_No response_

### 🌐 Protocol Configuration

_No response_

### 📋 Registry Configuration

_No response_

### 💾 Operating System

🍎 macOS

### 📝 Bug Description

../../../.gvm/pkgsets/go1.25.2/global/pkg/mod/github.com/cch123/supermonkey@v1.0.1/internal/bouk/replace.go:24:14: undefined: JmpToFunctionValue

### 🔄 Steps to Reproduce

go test ./pkg/server/... ./pkg/config/xds/...

### ✅ Expected Behavior

go test pass

### ❌ Actual Behavior

go test fail

### 💡 Possible Solution

_No response_

--- a/.gitignore
+++ b/.gitignore
@@ -33,4 +33,4 @@ target/
 samples/**/dist
 
 out/
-/test
+/test.worktrees/

--- a/go.mod
+++ b/go.mod
@@ -7,10 +7,10 @@ require (
 	dubbo.apache.org/dubbo-go/v3 v3.3.1
 	github.com/IBM/sarama v1.45.1
 	github.com/MicahParks/keyfunc v1.0.0
+	github.com/agiledragon/gomonkey/v2 v2.11.0
 	github.com/alibaba/sentinel-golang v1.0.4
 	github.com/apache/dubbo-getty v1.4.10
 	github.com/apache/dubbo-go-hessian2 v1.12.5
-	github.com/cch123/supermonkey v1.0.1
 	github.com/creasty/defaults v1.5.2
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/dubbogo/go-zookeeper v1.0.4-0.20211212162352-f9d2183d89d5
@@ -62,6 +62,7 @@ require (
 	go.uber.org/zap v1.21.0
 	golang.org/x/crypto v0.44.0
 	golang.org/x/net v0.47.0
+	golang.org/x/sync v0.18.0
 	google.golang.org/grpc v1.66.2
 	google.golang.org/protobuf v1.36.6
 	gopkg.in/yaml.v3 v3.0.1
@@ -251,7 +252,6 @@ require (
 	go.yaml.in/yaml/v3 v3.0.4 // indirect
 	golang.org/x/arch v0.20.0 // indirect
 	golang.org/x/mod v0.30.0 // indirect
-	golang.org/x/sync v0.18.0 // indirect
 	golang.org/x/sys v0.38.0 // indirect
 	golang.org/x/text v0.31.0 // indirect
 	golang.org/x/time v0.12.0 // indirect

--- a/pkg/config/xds/apiclient/grpc_test.go
+++ b/pkg/config/xds/apiclient/grpc_test.go
@@ -24,7 +24,7 @@ import (
 )
 
 import (
-	"github.com/cch123/supermonkey"
+	"github.com/agiledragon/gomonkey/v2"
 
 	"github.com/golang/mock/gomock"
 
@@ -130,17 +130,17 @@ func TestGRPCCluster_GetConnect(t *testing.T) {
 
 	gconn := &grpc.ClientConn{}
 	var state = connectivity.Ready
-	supermonkey.Patch(grpc.DialContext, func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+	patches := gomonkey.ApplyFunc(grpc.DialContext, func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
 		return gconn, nil
 	})
-	supermonkey.Patch((*grpc.ClientConn).Close, func(_ *grpc.ClientConn) error {
+	defer patches.Reset()
+	patches.ApplyMethod(&grpc.ClientConn{}, "Close", func(_ *grpc.ClientConn) error {
 		return nil
 	})
-	supermonkey.Patch((*grpc.ClientConn).GetState, func(_ *grpc.ClientConn) connectivity.State {
+	patches.ApplyMethod(&grpc.ClientConn{}, "GetState", func(_ *grpc.ClientConn) connectivity.State {
 		return state
 	})
 
-	defer supermonkey.UnpatchAll()
 	assert := require.New(t)
 	conn, err := g.GetConnection()
 	assert.NoError(err)

--- a/pkg/config/xds/cds_test.go
+++ b/pkg/config/xds/cds_test.go
@@ -23,7 +23,7 @@ import (
 )
 
 import (
-	"github.com/cch123/supermonkey"
+	"github.com/agiledragon/gomonkey/v2"
 
 	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 
@@ -98,12 +98,11 @@ func TestCdsManager_Fetch(t *testing.T) {
 	clusterMg := mocks.NewMockClusterManager(ctrl)
 	//var deltaResult chan *apiclient.DeltaResources
 	//var deltaErr error
-	supermonkey.Patch((*apiclient.GrpcExtensionApiClient).Fetch, func(_ *apiclient.GrpcExtensionApiClient, localVersion string) ([]*apiclient.ProtoAny, error) {
+	patches := gomonkey.ApplyMethod(&apiclient.GrpcExtensionApiClient{}, "Fetch", func(_ *apiclient.GrpcExtensionApiClient, localVersion string) ([]*apiclient.ProtoAny, error) {
 		return fetchResult, fetchError
 	})
-	//supermonkey.Patch(server.GetClusterManager, func() *server.ClusterManager {
-	//	return nil
-	//})
+	defer patches.Reset()
+
 	clusterMg.EXPECT().HasCluster(gomock.Any()).DoAndReturn(func(clusterName string) bool {
 		_, ok := cluster[clusterName]
 		return ok
@@ -140,7 +139,6 @@ func TestCdsManager_Fetch(t *testing.T) {
 	//supermonkey.Patch((*apiclient.GrpcExtensionApiClient).Delta, func(_ *apiclient.GrpcExtensionApiClient) (chan *apiclient.DeltaResources, error) {
 	//	return deltaResult, deltaErr
 	//})
-	defer supermonkey.UnpatchAll()
 
 	tests := []struct {
 		name              string

--- a/pkg/config/xds/xds_test.go
+++ b/pkg/config/xds/xds_test.go
@@ -24,7 +24,7 @@ import (
 )
 
 import (
-	monkey "github.com/cch123/supermonkey"
+	"github.com/agiledragon/gomonkey/v2"
 
 	"github.com/golang/mock/gomock"
 
@@ -69,39 +69,23 @@ func TestAdapter_createApiManager(t *testing.T) {
 
 	var state = connectivity.Ready
 	gconn := &grpc.ClientConn{}
-	monkey.Patch(grpc.DialContext, func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+	patches := gomonkey.ApplyFunc(grpc.DialContext, func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
 		fmt.Println("***** DialContext")
 		return gconn, nil
 	})
-	monkey.Patch((*grpc.ClientConn).Close, func(_ *grpc.ClientConn) error {
+	defer patches.Reset()
+
+	patches.ApplyMethod(&grpc.ClientConn{}, "Close", func(_ *grpc.ClientConn) error {
 		return nil
 	})
-	monkey.Patch((*grpc.ClientConn).GetState, func(_ *grpc.ClientConn) connectivity.State {
+	patches.ApplyMethod(&grpc.ClientConn{}, "GetState", func(_ *grpc.ClientConn) connectivity.State {
 		return state
 	})
-	//monkey.Patch(server.GetDynamicResourceManager, func() server.DynamicResourceManager {
-	//	return &server.DynamicResourceManagerImpl{}
-	//})
-	//monkey.Patch((*server.DynamicResourceManagerImpl).GetLds, func(_ *server.DynamicResourceManagerImpl) *model.ApiConfigSource {
-	//	return &apiConfig
-	//})
-	//monkey.Patch((*server.DynamicResourceManagerImpl).GetCds, func(_ *server.DynamicResourceManagerImpl) *model.ApiConfigSource {
-	//	return &apiConfig
-	//})
-	//monkey.Patch(server.GetClusterManager, func() *server.ClusterManager {
-	//	return nil
-	//})
-	//monkey.Patch((*server.ClusterManager).CloneStore, func(_ *server.ClusterManager) (*server.ClusterStore, error) {
-	//	return &server.ClusterStore{
-	//		Config:  []*model.Cluster{cluster},
-	//		Version: 1,
-	//	}, nil
-	//})
 
-	monkey.Patch((*apiclient.GrpcExtensionApiClient).Fetch, func(_ *apiclient.GrpcExtensionApiClient, localVersion string) ([]*apiclient.ProtoAny, error) {
+	patches.ApplyMethod(&apiclient.GrpcExtensionApiClient{}, "Fetch", func(_ *apiclient.GrpcExtensionApiClient, localVersion string) ([]*apiclient.ProtoAny, error) {
 		return nil, nil
 	})
-	monkey.Patch((*apiclient.GrpcExtensionApiClient).Delta, func(_ *apiclient.GrpcExtensionApiClient) (chan *apiclient.DeltaResources, error) {
+	patches.ApplyMethod(&apiclient.GrpcExtensionApiClient{}, "Delta", func(_ *apiclient.GrpcExtensionApiClient) (chan *apiclient.DeltaResources, error) {
 		ch := make(chan *apiclient.DeltaResources)
 		close(ch)
 		return ch, nil
@@ -123,8 +107,6 @@ func TestAdapter_createApiManager(t *testing.T) {
 		//clusterMg.EXPECT().HasCluster("cluster-2").Return(false)
 		apiclient.Init(clusterMg)
 	}
-
-	defer monkey.UnpatchAll()
 
 	ada := Xds{
 		clusterMg: clusterMg,

--- a/pkg/server/dynamic_resource_manager_test.go
+++ b/pkg/server/dynamic_resource_manager_test.go
@@ -25,7 +25,7 @@ import (
 )
 
 import (
-	"github.com/cch123/supermonkey"
+	"github.com/agiledragon/gomonkey/v2"
 
 	"github.com/stretchr/testify/require"
 )
@@ -175,13 +175,14 @@ func Test_createDynamicResourceManger(t *testing.T) {
 		},
 	}
 
-	supermonkey.Patch(xds.StartXdsClient, func(listenerMg controls.ListenerManager, clusterMg controls.ClusterManager, drm controls.DynamicResourceManager) xds.Client {
+	patches := gomonkey.ApplyFunc(xds.StartXdsClient, func(listenerMg controls.ListenerManager, clusterMg controls.ClusterManager, drm controls.DynamicResourceManager) xds.Client {
 		return nil
 	})
-	supermonkey.Patch((*Server).GetListenerManager, func(_ *Server) *ListenerManager {
+	defer patches.Reset()
+	patches.ApplyMethod(&Server{}, "GetListenerManager", func(_ *Server) *ListenerManager {
 		return nil
 	})
-	supermonkey.Patch((*Server).GetClusterManager, func(_ *Server) *ClusterManager {
+	patches.ApplyMethod(&Server{}, "GetClusterManager", func(_ *Server) *ClusterManager {
 		return nil
 	})
 

